hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanActionExecutor.java

```diff
@@ -205,31 +205,19 @@ private HoodieCleanMetadata runClean(HoodieTable<T, I, K, O> table, HoodieInstan
           Option.of(timer.endTimer()),
           cleanStats
       );
-      writeMetadata(metadata);
+      if (!skipLocking && config.isMetadataTableEnabled()) {
```

Comment:
can we have a `needsLocking` member which is initialized to `needsLocking = !skipLocking && config.isMetadataTableEnabled()` and just reuse here and below?